server/services/team/service.ts

```diff
@@ -211,17 +215,29 @@ export class TeamService extends ServiceClass implements ITeamService {
 		if (!uid) {
 			throw new Error('missing-userId');
 		}
-		const room = await this.RoomsModel.findOneByRoomIdAndUserId(rid, uid);
+		if (!canRemoveAnyRoom) {
+			const room = await this.RoomsModel.findOneById(rid);
+			const user = await this.Users.findOneById(uid);
+			const canSeeRoom = await canAccessRoom(room, user);
+			if (!canSeeRoom) {
+				throw new Error('user-not-member-of-room');
+			}
+		}
+
+		const room = await this.RoomsModel.findOneById(rid);
```

Comment:
```suggestion
		const room = await this.RoomsModel.findOneById(rid);
		if (!canRemoveAnyRoom) {
			const user = await this.Users.findOneById(uid);
			const canSeeRoom = await canAccessRoom(room, user);
			if (!canSeeRoom) {
				throw new Error('user-not-member-of-room');
			}
		}
```

The room will be loaded anyway, so it's better to load it only once before the checks.